command/base.go

```diff
@@ -321,12 +327,24 @@ func (c *BaseCommand) flagSet(bit FlagSetBit) *FlagSets {
 				Completion: complete.PredictAnything,
 				Usage:      "Address of the Vault server.",
 			}
+
+			flagAddrWarning := &StringVar{
+				Name:    "addr-warning",
+				Target:  &c.flagAddrWarning,
+				Default: "",
+				Hidden:  true,
+				Usage:   "For displaying: WARNING! VAULT_ADDR and -address unset. Defaulting to https://127.0.0.1:8200.",
+			}
+
 			if c.flagAddress != "" {
 				addrStringVar.Default = c.flagAddress
 			} else {
 				addrStringVar.Default = "https://127.0.0.1:8200"
+				flagAddrWarning.Default = "WARNING! VAULT_ADDR and -address unset. Defaulting to %s."
```

Comment:
I don't think you need StringVar here.  I would just say instead here: `c.flagAddrWarning = fmt.Sprintf("WARNING! VAULT_ADDR and -address unset. Defaulting to %s.", addrStringVar.Default)`.